<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>740</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0" rowspan="4">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string notr="true"/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPushButton" name="pbDirList">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>ImageSource</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbWatchlist">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>WindowWatch</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbStayOnTop">
         <property name="text">
          <string>StayOnTop</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="dg::rectwindow" name="rectw" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Controls</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="3" column="0" colspan="2">
        <widget class="dg::CurIndicator" name="numIndicator">
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>15</pointsize>
           <weight>50</weight>
           <italic>true</italic>
           <bold>false</bold>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="lineWidth">
          <number>2</number>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QPushButton" name="pbInit">
         <property name="text">
          <string comment="Sprinkle">Reset</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QPushButton" name="pbNext">
         <property name="text">
          <string>Sprinkle</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QProgressBar" name="progressBar">
         <property name="value">
          <number>0</number>
         </property>
         <property name="textVisible">
          <bool>true</bool>
         </property>
         <property name="invertedAppearance">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QPushButton" name="pbCurrent">
         <property name="text">
          <string>Re-Sprinkle</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QListView" name="listKeep">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="viewMode">
          <enum>QListView::ListMode</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Keep</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="removeKeep">
         <property name="text">
          <string comment="Keep">Remove</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" rowspan="3">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>ImageCollect-Constraint</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="3" column="0" colspan="2">
        <widget class="dg::MySliderD" name="slMin" native="true"/>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="dg::MySliderI" name="slSamp" native="true"/>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="dg::MySliderD" name="slMax" native="true"/>
       </item>
       <item row="0" column="0">
        <widget class="dg::MyCheckbox" name="cbPreferOriginal">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>PreferOriginal</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>740</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMenu">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="actionOpenWatchList"/>
    <addaction name="actionOpenDirList"/>
    <addaction name="actionStayOnTop"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpenWatchList">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ImageSource</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionStayOnTop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>StayOnTop</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionOpenDirList">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>WindowWatch</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionSprinkle">
   <property name="text">
    <string>Sprinkle</string>
   </property>
  </action>
  <action name="actionRe_Sprinkle">
   <property name="text">
    <string>Re-Sprinkle</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>dg::rectwindow</class>
   <extends>QWidget</extends>
   <header>src/rectwindow.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>dg::CurIndicator</class>
   <extends>QLabel</extends>
   <header>src/cur_indicator.hpp</header>
   <slots>
    <slot>onSprinkleCounterChanged(size_t,size_t)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>dg::MyCheckbox</class>
   <extends>QCheckBox</extends>
   <header>src/mycheckbox.hpp</header>
  </customwidget>
  <customwidget>
   <class>dg::MySliderI</class>
   <extends>QWidget</extends>
   <header>src/myslider_i.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>dg::MySliderD</class>
   <extends>QWidget</extends>
   <header>src/myslider_d.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpenWatchList</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showWatchList(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpenWatchList</sender>
   <signal>toggled(bool)</signal>
   <receiver>pbWatchlist</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>116</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbWatchlist</sender>
   <signal>clicked()</signal>
   <receiver>actionOpenWatchList</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbStayOnTop</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionStayOnTop</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStayOnTop</sender>
   <signal>toggled(bool)</signal>
   <receiver>cbStayOnTop</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStayOnTop</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>stayOnTop(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpenDirList</sender>
   <signal>toggled(bool)</signal>
   <receiver>pbDirList</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpenDirList</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showDirList(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbDirList</sender>
   <signal>clicked()</signal>
   <receiver>actionOpenDirList</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>276</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbInit</sender>
   <signal>clicked()</signal>
   <receiver>actionReset</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>458</y>
    </hint>
    <hint type="destinationlabel">
     <x>616</x>
     <y>361</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbNext</sender>
   <signal>clicked()</signal>
   <receiver>actionSprinkle</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>718</x>
     <y>458</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>sprinkleCounterChanged(size_t,size_t)</signal>
   <receiver>numIndicator</receiver>
   <slot>onSprinkleCounterChanged(size_t,size_t)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>718</x>
     <y>404</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>onProgress(int)</signal>
   <receiver>progressBar</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>374</y>
    </hint>
    <hint type="destinationlabel">
     <x>718</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbCurrent</sender>
   <signal>clicked()</signal>
   <receiver>actionRe_Sprinkle</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>718</x>
     <y>485</y>
    </hint>
    <hint type="destinationlabel">
     <x>444</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeKeep</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeKeep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>509</x>
     <y>347</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSprinkle</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sprinkle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>369</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRe_Sprinkle</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>reSprinkle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>369</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sprinkleReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>369</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>sprinkleCounterChanged(size_t,size_t)</signal>
  <signal>onProgress(int)</signal>
  <slot>showWatchList(bool)</slot>
  <slot>stayOnTop(bool)</slot>
  <slot>showDirList(bool)</slot>
  <slot>sprinkle()</slot>
  <slot>sprinkleReset()</slot>
  <slot>reSprinkle()</slot>
  <slot>removeKeep()</slot>
 </slots>
</ui>
